driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -211,18 +266,27 @@ public static <T extends Expression> MapExpression<T> ofMap() {
     }
 
     /**
-     * user asserts type of values is T
+     * Returns a {@linkplain MapExpression map} value corresponding to the
+     * provided {@link Bson Bson document}. The user asserts that all values
```

Comment:
The prose style and text formatting is different in `Expression.isArrayOr`/`isMapOr`. Shouldn't we use the same everywhere? I am not saying what it should be.